discord/sink.py

```diff
@@ -0,0 +1,775 @@
+        # Tested this and it appears that discord does not send
+        # the specified amount of report blocks
+        # fmt = ">IB3s4I"
+        # buf_size = struct.calcsize(fmt)
+        # report_blocks = [struct.unpack_from(fmt, buffer=data, offset=buf_size * i)
+        #                  for i in range(self.rc)]
+        # self.report_blocks = list(map(lambda args: RTCPReceiverReportBlock(
+        #     *args[:2], int.from_bytes(args[2], 'big'), *args[3:]
+        # ), report_blocks))
+        self.report_blocks = []
+
+        self.extension = data  # data[len(self.report_blocks) * buf_size:]
+
+
+class RTCPSenderReportPacket(RTCPPacket):
+    """RTCP Sender Report packet which provides quality feedback
+
+    Read in detail here: https://www.freesoft.org/CIE/RFC/1889/19.htm
+
+    Extends :class:`RTCPPacket` and inherits its attributes.
+
+    Attributes
+    ----------
+    ssrc: :class:`int`
+        The synchronization source identifier for the originator of this SR packet.
+    nts: :class:`int`
+        NTP timestamp. Indicates the wallclock time when this report was sent
+        so that it may be used in combination with timestamps returned in
+        reception reports from other receivers to measure round-trip
+        propagation to those receivers.
+    rts: :class:`int`
+        RTP timestamp. Corresponds to the same time as the NTP timestamp (above),
+        but in the same units and with the same random offset as the RTP
+        timestamps in data packets.
+    spc: :class:`int`
+        The total number of RTP data packets transmitted by the sender since
+        starting transmission up until the time this SR packet was generated.
+        The count is reset if the sender changes its SSRC identifier.
+    soc: :class:`int`
+        The total number of payload octets (i.e., not including header or padding)
+        transmitted in RTP data packets by the sender since starting transmission
+        up until the time this SR packet was generated. The count is reset if
+        the sender changes its SSRC identifier.
+    report_blocks: Sequence[:class:`RTCPReceiverReport`]
+        Sequence of :class:`RTCPReceiverReport` objects that tell statistics.
+        Receivers do not carry over statistics when a source changes its SSRC
+        identifier due to a collision.
+    extension: :class:`bytes`
+        Profile-specific extension that may or may not contain a value.
+    """
+    __slots__ = RTCPPacket.__slots__ + (
+        "ssrc", "nts", "rts", "spc", "soc", "report_blocks",
+        "extension"
+    )
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        fmt = ">IQ3I"
+        buf_size = struct.calcsize(fmt)
+        self.ssrc, self.nts, self.rts, self.spc, self.soc = \
+            struct.unpack(fmt, data[:buf_size])
+
+        self._parse_report_and_extension(data[buf_size:])
```

Comment:
Same deal as other suggestion...

```suggestion
        self.ssrc, self.nts, self.rts, self.spc, self.soc = \
            struct.unpack_from("!IQ3I, data, 0)
        self.report_blocks = []
        self.extension = data[:24]
```